ddclient.in

```diff
@@ -7162,6 +7175,65 @@ sub nic_keysystems_update {
   }
 }
 
+######################################################################
+## nic_regfishdev4_examples
+######################################################################
+sub nic_regfishdev4_examples {
+    return <<EoEXAMPLE;
+o 'regfishde'
+The 'regfishde' protocol is used by the non-free dynamic DNS service offered by www.regfish.de.
+Check https://www.regfish.de for configuration variables applicable to the 'regfishde' protocol:
+protocol=regfishde
+server=dyndns.regfish.de
+password=service-password          ## password (token) registered with the service
+myhost.mydomain.com                ## the host registered with the service.
+Example ${program}.conf file entries:
+## single host update
+protocol=regfishde, \
+password=service-password \
+myhost.mydomain.com
+EoEXAMPLE
+}
+
+######################################################################
+## nic_regfishdev4_update
+## response contains "success" on succesfull completion
+######################################################################
+sub nic_regfishdev4_update {
+ debug("\nnic_regfishdev4_update -------------------");
+
+ ## update each configured host
+ ## should improve to update in one pass
+ for my $h (@_) {
+   my $ip = delete $config{$h}{'wantip'};
+   info("regfish.de setting IP address to %s for %s", $ip, $h);
+
+   my $url = "https://$config{$h}{'server'}/?fqdn=$h&ipv4=$ip&forcehost=1&token=$config{$h}{'password'}";
+
+   # Try to get URL
+   my $reply = geturl(proxy => opt('proxy'), url => $url);
+
+   # No response, declare as failed
+   if (!defined($reply) || !$reply) {
+     failed("regfish.de updating %s: Could not connect to %s.", $h, $config{$h}{'server'});
```

Comment:
Could not connect is a very specific failure and this call can fail for many other reasons. This should not be generalized here.

Reply:
So what other should i do?
I'm sorry but i'm not really familary with this.

Reply:
```suggestion
     failed("regfish.de updating %s failed: %s.", $h, $config{$h}{'server'});
```

Reply:
You're right, thats better.